### ISSUE TITLE
OpenMP Parallelisation Problem

### ISSUE THREAD

**Issue by alxstan** [tblite/tblite#202]
Hi, I have switched from `xtb-python` to `tblite` to use the xTB Python interface. However, when trying to compute singlepoints for the same molecular systems as done with `xtb-python` I noticed that there is a decrease in performance when increasing the number of threads, which was not the case for the former. For example, setting `OMP_NUM_THREADS` to 64 instead of 8 makes my calculation 1.3 times slower. I tested different amounts of threads on the same system and made sure that the threads are also used as set, but I see a consistent decrease in performance regardless of the example.

![walltimes](https://github.com/user-attachments/assets/a8c53db6-57c9-4de9-9910-bbccc79150cf)
These are results for GFN2-xTB singlepoints for a molecular system consisting of 140 atoms and using tblite during an MD. I averaged over at least 500 000 data points (as I did some IQR outlier cleaning) so I do not really understand what could cause such behavior. It also looks really wrong to me.

Is the problem known? Or am I doing something wrong?

**Comment by pitsteinbach** [comment]
Hey,
yes it is known that there is a diminishing return for parallelization.

I previously observed this behavior mainly for binaries linked against OpenBLAS as opposed to intel MKL.
Could you specify which BLAS library you are using and also the OMP library used?

**Comment by alxstan** [comment]
Hi! Sorry for the late response, we recompiled `tblite` with Intel MKL and now the scaling behavior with the number of threads has returned back to normal as it was with `xtb-python`. Thank you for the tip!

Anyway, here are the libraries we used before:
 libopenblas.so.0 => .../conda-venvs/3.9/latest/env/lib/libopenblas.so.0 (0x00007f02c6509000)
 libgomp.so.1 => .../conda-venvs/3.9/latest/env/lib/libgomp.so.1 (0x00007f02c620d000)
